### PR TITLE
Add types for Client.queryWithMetrics

### PR DESCRIPTION
**Pull request by ptpaterson**
#625 did not add types for the Client class.

### CODE REVIEW

File: src/types/Client.d.ts [hunk 1]
```diff
@@ -49,9 +49,24 @@ interface StreamApi
...
+  }
+}
+
 export default class Client {
   constructor(opts?: ClientConfig)
   query<T = object>(expr: ExprArg, options?: QueryOptions): Promise<T>
+  queryWithMetrics<T = object>(
+    expr: ExprArg,
+    options?: QueryOptions
+  ): Promise<MetricsResponse<T>>
```
Comment (doom-weaver) [inline src/types/Client.d.ts]: query now returns an or type.

: `Promise<T> | Promise<MetricsResponse<T>>`

**Review by doom-weaver — DISMISSED** [review]

File: src/types/Client.d.ts [hunk 2]
```diff
@@ -49,9 +49,24 @@ interface StreamApi
...
+    'x-byte-read-ops': number
+    'x-byte-write-ops': number
+    'x-query-time': number
+    'x-txn-retries': number
+  }
+}
+
 export default class Client {
   constructor(opts?: ClientConfig)
   query<T = object>(expr: ExprArg, options?: QueryOptions): Promise<T>
```
Comment (doom-weaver) [inline src/types/Client.d.ts]:
```suggestion
  query<T = object>(expr: ExprArg, options?: QueryOptions): Promise<T> | Promise<MetricsResponse<T>>
```

We need to use a Union type here for the case where the user explicitly passes the `metrics: true` option

**Review by doom-weaver — APPROVED** [review]

File: src/types/Client.d.ts [hunk 3]
```diff
@@ -49,9 +49,24 @@ interface StreamApi
...
+    'x-byte-write-ops': number
+    'x-query-time': number
+    'x-txn-retries': number
+  }
+}
+
 export default class Client {
   constructor(opts?: ClientConfig)
-  query<T = object>(expr: ExprArg, options?: QueryOptions): Promise<T>
+  query<T = object>(expr: ExprArg, options?: QueryOptions): Promise<T> | Promise<MetricsResponse<T>>
```
Comment (Akryum) [inline src/types/Client.d.ts:65]: This is a breaking change for TypeScript projects :(

Comment (Akryum) [reply src/types/Client.d.ts:65]: Example:

image

```
error TS2322: Type 'boolean | MetricsResponse<boolean>' is not assignable to type 'boolean'.
```

**Pull request merged**